File: go/add-border/add-border.go
package main

import (
	"fmt"
	"strings"
)

func addBorder(picture []string)[]string{
	character := "*"
	width := len(picture[0])
	topBottomBorder := strings.Repeat(character, width+2)

	res := []string{}


	for i := 0 ; i < len(picture) ; i++{
		res = append(res, character+picture[i]+character)
	} 

	res = append([]string{topBottomBorder}, res...)
	res = append(res, topBottomBorder)
	return res
}

func main() {
	picture := []string{"abc", "ded"}
	fmt.Printf("The answer for the array %v is %v", picture, addBorder(picture))
}

File: js/addBorder.js
function addBorder(picture){
    let character='*'
    let width=picture[0].length
    let topBottomBorder=character.repeat(width+2)

    let partialArray=[]

    for(i=0;i<picture.length;i++){
        partialArray.push(character+picture[i]+character)
    }

    partialArray.unshift(topBottomBorder)
    partialArray.push(topBottomBorder)

    return partialArray
}

let picture = ["abc", "ded"]
console.log(addBorder(picture))